File: hypertube-frontend/src/app/models/movie.model.ts
import { Time } from "@angular/common";

export interface Movie {
  id: number;
  title: string;
  englishTitle: string;
  overview: string;
  imdb_id: string;

  vote_average: number;
  release_date: string;
  poster_path: string;
  backdrop_path: string;

  runtime: number;
  genres: GenreModel[];
  credits: Credits;
  subtitles: Map<string, string>;

  stoppedAt: Time;
}

export interface MovieThumbnail {
  id: number;
  title: string;

  release_date: number;
  poster_path: string;
  runtime: number;
  // stoppedAt: Time;
  stoppedAt: string;
  genres: GenreModel[];
  vote_average: number;
}

export interface Subtitles {
  title: string,
  language: string,
  url: string
}
export interface GenreModel {
    id: number;
    name: string;
}

export interface PersonModel {
  id: number;
  name: string;
  character?: string;
  job?: string;
  profile_path?: string;
}

export interface WatchedMoviesDTO {
  userId: number;
  movieId: number;
  watchedAt: string;
}

  export interface Credits {
    cast: PersonModel[];
    crew: PersonModel[];
  }

  export interface MovieDTO {
    id: number;
    title: string;
    overview: string;
    rating: number;
    releaseDate: string;
    thumbnail: string;
    genreIds: number[];
    runtime: number;
    genres: GenreModel[];
    credits: Credits;
    subtitles: Record<string, string>;
    watchedMovies: WatchedMoviesDTO;
  }

  export interface SearchMovie {
    query: string;
    page: number;
    genresIds: number[];
  }

File: core-api/src/main/java/com/hypertube/core_api/model/MovieModel.java
package com.hypertube.core_api.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.sql.Time;
import java.util.List;
import java.util.Map;

@JsonIgnoreProperties(ignoreUnknown = true)
public class MovieModel {

    private Integer id;
    private String title;
    private String overview;

    @JsonProperty("imdb_id")
    private String imdbId;

    @JsonProperty("vote_average")
    private Double voteAverage;

    @JsonProperty("release_date")
    private String releaseDate;

    @JsonProperty("poster_path")
    private String thumbnail;

    @JsonProperty("backdrop_path")
    private String backdropPath;

    @JsonProperty("genre_ids")
    private List<Integer> genreIds;

    private Integer runtime;
    private List<GenreModel> genres;
    private Credits credits;
    private Map<String, String> subtitles;

    private Time stoppedAt;

    private String englishTitle;

    public static class Credits {
        public List<PersonModel> cast;
        public List<PersonModel> crew;
    }

    public Credits getCredits() {
        return credits;
    }

    public void setCredits(Credits credits) {
        this.credits = credits;
    }

    public List<Integer> getGenreIds() {
        return genreIds;
    }

    public void setGenreIds(List<Integer> genreIds) {
        this.genreIds = genreIds;
    }

    public List<GenreModel> getGenres() {
        return genres;
    }

    public void setGenres(List<GenreModel> genres) {
        this.genres = genres;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getOverview() {
        return overview;
    }

    public void setOverview(String overview) {
        this.overview = overview;
    }

    public Double getVoteAverage() {
        return voteAverage;
    }

    public void setVoteAverage(Double voteAverage) {
        this.voteAverage = voteAverage;
    }

    public String getReleaseDate() {
        return releaseDate;
    }

    public void setReleaseDate(String releaseDate) {
        this.releaseDate = releaseDate;
    }

    public Integer getRuntime() {
        return runtime;
    }

    public void setRuntime(Integer runtime) {
        this.runtime = runtime;
    }

    public Map<String, String> getSubtitles() {
        return subtitles;
    }

    public void setSubtitles(Map<String, String> subtitles) {
        this.subtitles = subtitles;
    }

    public String getThumbnail() {
        return thumbnail;
    }

    public void setThumbnail(String thumbnail) {
        this.thumbnail = thumbnail;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Time getStoppedAt() {
        return stoppedAt;
    }

    public void setStoppedAt(Time stoppedAt) {
        this.stoppedAt = stoppedAt;
    }

    public String getImdbId() {
        return imdbId;
    }

    public void setImdbId(String imdbId) {
        this.imdbId = imdbId;
    }

    public String getBackdropPath() {
        return backdropPath;
    }

    public void setBackdropPath(String backdropPath) {
        this.backdropPath = backdropPath;
    }

    public String getEnglishTitle() {
        return englishTitle;
    }

    public void setEnglishTitle(String englishTitle) {
        this.englishTitle = englishTitle;
    }
}
